buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/TapLoggerOutputStream.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.vagrant;
+
+import org.elasticsearch.gradle.LoggingOutputStream;
+import org.gradle.api.GradleScriptException;
+import org.gradle.api.logging.Logger;
+import org.gradle.internal.logging.progress.ProgressLogger;
+
+import java.util.Formatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Adapts an OutputStream containing output from bats into a ProgressLogger
+ * and a Logger. Every test output goes to the ProgressLogger and all failures
+ * and non-test output goes to the Logger. That means you can always glance
+ * at the result of the last test and the cumulative pass/fail/skip stats and
+ * the failures are all logged.
+ *
+ * There is a Tap4j project but we can't use it because it wants to parse the
+ * entire TAP stream at once and won't parse it stream-wise.
+ */
+public class TapLoggerOutputStream extends LoggingOutputStream {
```

Comment:
I had to do some googling to find out what "tap" is. Can we perhaps elaborate here in the javadoc on that, and even link to https://testanything.org?